Salesforce Connector Studio Topic to Edit (#614)
mulesoft/docs-connectors: modules/ROOT/pages/salesforce/salesforce-connector-100-studio.adoc

. Drag a connector operation to the Studio canvas.
+

> To use the Salesforce connector as an input source, drag one of these operations to the Source area of a flow. A channel or topic must be published to Salesforce before a subscription to the channel can be created.

Reviewer: ```suggestion
To use Salesforce Connector as an input source, drag one of these operations to the Source area of a flow. A channel or topic must be published to Salesforce before a subscription to the channel can be created.
```
Do you want to leave the second sentence in passive voice?

+
* *On Deleted Object*
+
Initiates your app when a Salesforce user deletes an object. Possible object types are `Contact`, `Lead`, `Account`, `Contracts`, `Opportunity`, or the name of a custom object.
+
* *On Modified Object*
+
Initiates your app when a Salesforce user modifies an object. Possible object types are `Contact`, `Lead`, `Account`, `Contracts`, `Opportunity`, or the name of a custom object.
+
* *On New Object*
Initiates your app when a Salesforce user creates a new object. Possible object types are `Contact`, `Lead`, `Account`, `Contracts`, `Opportunity`, or the name of a custom object.
* *Replay channel listener*
+
Subscribes to a streaming channel. This input source provides channel-related notification settings for new events and events stored by Salesforce within the 24-hour retention period. Events that take place in the specified channel input source the flow.
+

> * *Replay topic listener*

Reviewer: I have been assuming but should probably double-check: the UI labels use inconsistent capitalization?

+

> Subscribes to a topic. This input source provides topic-related notification settings for new events and events stored by Salesforce within the 24-hour retention period. Events that take place on records in the specified topic trigger the flow.

Reviewer: Ugh, I didn't notice this in the previous PR edit, but it's a good idea to replace "take place" with "occur," for localization/Global English.

+
* *Subscribe channel listener*
+

> Subscribes to a streaming channel. This input source provides channel-related notification settings for new events that take place after you subscribe.

Reviewer: ditto
If you agree to change this, just search and evaluate and replace.

+
* *Subscribe topic listener*
+
Subscribe to a topic. This input source provides topic-related notification settings for new events that take place after you subscribe.
+
For the *On New Object* and *On Modified Object* operations, specify a UTC value in the *Since* field. If the server is in a different time zone, you must specify a value in this field. If you leave the field empty, the connector uses the local timestamp.
+
. Click the operation.
. Click the green plus sign to the right of *Connector Configuration*.

> . In the *Connection* field choose one of the following connection types:

Reviewer: ```suggestion
. In the *Connection* field, choose one of the following connection types:
```

+
. For each of the connection choices, you can set optional proxy values:
+
** *Host*
+
Host name of the proxy server
+
** *Port*
+

> Port number on which the proxy server runs

Reviewer: ```suggestion
Number of the port on which the proxy server runs
```

+
** *Username*
+
Username to log in to the server
+
** *Password*
+
Corresponding password

> After completing fields in the *Global Element Properties* screen, return to the main connector menu and set the <<Connector Property Values>>.

Reviewer: "all of the"?
```suggestion
After completing the fields in the *Global Element Properties* screen, return to the main connector menu and set the <<Connector Property Values>>.
```

== Basic Authentication
Important fields:

* *Username* (required)
+
Salesforce username
+
* *Password* (required)
+
Corresponding password
* *Security token*
+
Corresponding security token
+
* *Authorization URL*
+

> URL for the endpoint configured to handle SOAP authentication requests

Reviewer: "handle" is another of those terms that can confuse nonnative readers. Consider replacing with "manage" or "process" or similar.

+
** *External callback url*
+
Callback URL, for example, `+http://localhost:8085/callback+`

== OAuth JWT

Important fields:

* *Consumer key* (required)
+
Consumer key for the Salesforce-connected app. See <<Create a Consumer Key>>.
+
* *Key store* (required)
+
See <<Generate a Keystore File>>.
+
* *Store password* (required)
+
Password for the keystore
+
* *Principal* (required)
+
Username of the Salesforce user

== OAuth SAML

Important fields:

* *Consumer key* (required)
+
Consumer key for the Salesforce-connected app. See <<Create a Consumer Key>>.
+
* *Key store* (required)
+
See <<Generate a Keystore File>>.
+
* *Store password* (required)
+
Password for the keystore
+
* *Principal* (required)
+
Username of the Salesforce user

== OAuth Username Password

Important fields:

* *Consumer key* (required)
+
Consumer key for the Salesforce-connected app. See <<Create a Consumer Key>>.
+
* *Consumer secret* (required)
+
Consumer secret for the connector to access Salesforce
+
* *Username* (required)
+
Salesforce username
+
* *Password* (required)
+
Corresponding password
+
* *Security token*
+
Corresponding security token

[[propvals]]
== Connector Property Values

> The following example shows four operations of the many you can set for the Salesforce connector, as well as the important fields for these operations:

Reviewer: ```suggestion
The following example shows four operations of the many you can set for Salesforce Connector, as well as the important fields for these operations:
```

This procedure assumes that you already have a certification file (such as `salesforce-cert.crt`). If not, you can produce one by generating a Java KeyStore and Public Key.

[[create-consumer-key]]
. Log in to Salesforce and go to *Setup* > *Build* > *Create* > *Apps*.
. Under the Connected App section, click *New*.
. To create a new connected app, enter:
+
* A name for the connected app
* The API name
* The contact email
+
. Under API (Enable OAuth Settings), select *Enable OAuth Settings*:
+
* Enter the *Callback URL*.
* Select the *Use Digital Signatures* checkbox.
* Click *Browse* (or *Choose File*), and load your Salesforce certificate (for example, `salesforce-cert.crt`), which contains your public key.
+
In Studio, you typically store the certificate in the workspace that contains your Mule app.
+
. Add and save these OAuth scopes to *Selected OAuth Scopes*:
+
** *Full Access* (`full`)
** *Perform Requests On Your Behalf At Any Time* (`refresh_token`, `offline_access`)
+
. Configure the authorization settings for the app.
. Click *Manage*.

> . In the OAuth Policies section, expand the *Permitted Users* dropdown, and select *Admin Approved Users are Pre-Authorized*.

Reviewer: list or menu?
```suggestion
. In the OAuth Policies section, expand the *Permitted Users* list and select *Admin Approved Users are Pre-Authorized*.
```

+
. Click *Save*.
. In the Profiles section, click *Manage Profiles*.
. Select your user profile and click *Save*.

> . Select *Build* > *Create* > *Apps* to go back to the list of connected apps.

Reviewer: "go back to" could be "return to" (localization)

keytool -genkeypair -alias salesforce-cert -keyalg RSA -keystore salesforce-cert.jks
----
+

> . Enter the following Information:

Reviewer: ```suggestion
. Enter the following information:
```

+
** Password for the keystore
** Your first name and last name
** Your organization unit

> ** Name of your city, state, and the two letters code of your country

Reviewer: ```suggestion
** Name of your city, state, and the two-letter code for your country
```

+

> The system generates a Java keystore file in JKS format that contains a private or public key pair in your workspace.

Reviewer: Would a Java keystore file be in any other format besides JKS?

+
The system exports the public key from the keystore into the workspace. This is the public key that you need to enter in your Salesforce instance.
+

> . Ensure that you have both the keystore (`salesforce-cert.jks`) and the public key (`salesforce-cert.crt`) files in your workspace.

Reviewer: Verify?

== Use Mutual TLS

> In v9.7.0 and later, all authentication types support Mutual TLS. To use Mutual TLS, you need a keystore file in the JKS format and a password for the file. See <<Generate a Keystore File>> for more information.

Reviewer: Should we name the thing that is in version 9.7.0 and later?

Reviewer: GLOBALLY change "Mutual TLS" to "mutual TLS."
I looked this up and added it to our Term Presentation List.

> Specify the path to the keystore file and the password in the configuration window (as shown in the image below). Any user requiring Mutual TLS authentication can log in using the connector.

Reviewer: Consider this format:
```suggestion
Specify the path to the keystore file and the password in the configuration window:
<image>
Any user requiring mutual TLS authentication can log in using the connector.
```
or
```suggestion
Specify the path to the keystore file and the password in the configuration window (as shown in the following image). Any user requiring mutual TLS authentication can log in using the connector.
```

image::salesforce/salesforce-100-mutual-tls.png[Mutual TLS]

> To set up Mutual TLS certificate in your Salesforce environment, see https://help.salesforce.com/articleView?id=security_keys_uploading_mutual_auth_cert.htm&type=5[Set Up a Mutual Authentication Certificate].

Reviewer: ```suggestion
To set up a mutual TLS certificate in your Salesforce environment, see https://help.salesforce.com/articleView?id=security_keys_uploading_mutual_auth_cert.htm&type=5[Set Up a Mutual Authentication Certificate].
```

[[apexsets]]
== Apex Settings

> You can configure Apex REST and SOAP access using Studio or in XML. When you connect to Salesforce, the Salesforce connector gets the names of the Apex classes and methods belonging to them that can be invoked.

Reviewer: ```suggestion
You can configure Apex REST and SOAP access using Studio or in XML. When you connect to Salesforce, Salesforce Connector gets the names of the Apex classes, and methods belonging to them, that can be invoked.
```

All Salesforce connection configurations support these Apex settings:

* *Fetch All Apex SOAP Metadata*
+
Fetches the metadata of all the Apex SOAP classes. Takes precedence over Apex Class Name settings.
+
* *Fetch All Apex REST Metadata*
+
Fetches the metadata of all the all Apex REST classes. Takes precedence over Apex Class Name settings.
+
* *Apex Class Names*
+

> List of Apex class names to use for limiting the set of classes you fetch along with the methods they expose. This setting can speed up the fetch process if there are a lot of classes that you do not need to fetch.

Reviewer: ```suggestion
List of Apex class names to use for limiting the set of classes you fetch, along with the methods they expose. This setting can accelerate the fetch process if there are many classes that you do not need to fetch.
```

You can provide Apex settings in Design Center and in Studio 7.
See also xref:salesforce/salesforce-connector-xml-maven.adoc#apexml[Apex XML Settings].

Click the *Apex* tab to choose the settings.

Apex settings values:

* *Fetch All Apex SOAP Metadata*
+
Fetches the metadata of all the Apex SOAP classes
+
* *Fetch All Apex REST Metadata*
+
Fetches the metadata of all the all Apex REST classes

Apex Class Names:

The *Expression*, *Edit inline*, or *Bean reference* choices provide these options:

* *None*
+
No Apex class name is mentioned for DataSense to acquire.
* *From a message*
+
Enables you to specify the class name using an expression.
+
* *Create object manually*
+

> Enables you to create a list of class names for DataSense to acquire. DataSense will acquire only those classes and their associated methods.

Reviewer: ```suggestion
Enables you to create a list of class names for DataSense to acquire. DataSense acquires only those classes and their associated methods.
```

> The *Fetch All Apex SOAP Metadata* and *Fetch All Apex REST Metadata* check boxes take precedence over the *Apex Class Names* setting. If these boxes are selected, they fetch all the Apex SOAP metadata or Apex REST metadata, regardless of your selection in the Apex Class Names section.

Reviewer: ```suggestion
The *Fetch All Apex SOAP Metadata* and *Fetch All Apex REST Metadata* checkboxes take precedence over the *Apex Class Names* setting. If these boxes are selected, they fetch all the Apex SOAP metadata or Apex REST metadata, regardless of your selection in the Apex Class Names section.
```
